src/Data/Foreign/Internal.purs

```diff
@@ -0,0 +1,16 @@
+module Data.Foreign.Internal where
+
+import Prelude
+
+import Data.Foreign (F, Foreign, ForeignError(..), fail, isArray, tagOf, unsafeFromForeign)
+import Data.StrMap (StrMap)
+
+-- | Test whether a foreign value is a dictionary
+isStrMap :: Foreign -> Boolean
+isStrMap v = tagOf v == "Object" && not (isArray v || tagOf v == "Date")
```

Comment:
I have also attempted encoding/decoding Data.Map and it depends on this. Should I commit it in this PR or use another one?